### ISSUE TITLE
Redmine 4.0 support

### ISSUE THREAD

**Issue by octoquad** [makotokw/redmine-theme-gitmike#41]
Hi,

Are there any updates planned for Redmine 4.0?

Thanks

**Comment by makotokw** [comment]
Hi, I have been used my theme with Redmine 4.0 for a month. I never found any big issue.

If you know, please let me know what should do support for Redmine 4.0.

**Comment by octoquad** [comment]
Sure, I'll compile a small list and supply them to you, and provide you with some CSS changes I did a few months back.

**Comment by octoquad** [comment]
So, I had a look through our production system, and yeah, there are no breaking changes that I can see.

The diff below was to fix white text on a white background when Firefox added support to detect dark themes from Linux desktops a few releases ago. This is no longer needed as they properly fixed it in the last release, but perhaps you can use it to ensure that the text color is set.

```diff
diff --git a/stylesheets/application.css b/stylesheets/application.css
index fdf0d37..b2ac4a7 100644
--- a/stylesheets/application.css
+++ b/stylesheets/application.css
@@ -188,8 +188,8 @@ input[type="button"]:hover, input[type="submit"]:hover, input[type="reset"]:hove
 .btn.active, .btn:active { background-image: none; outline: 0; box-shadow: inset 0 2px 4px rgba(0, 0, 0, 0.15), 0 1px 2px rgba(0, 0, 0, 0.05); }
 .btn.disabled, .btn[disabled] { cursor: default; background-image: none; opacity: .65; box-shadow: none; }
 
-input[type="text"], input[type="password"], input[type="email"], input[type="number"], input[type="tel"], input[type="url"], textarea, select { outline: none; padding: 3px 4px; background-color: #fff; border: 1px solid #ccc; box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.075); border-radius: 3px; -webkit-transition: all 0.15s ease-in 0s; transition: all 0.15s ease-in 0s; vertical-align: middle; }
-#content input[type="text"], #content input[type="password"], #content input[type="email"], #content input[type="number"], #content input[type="tel"], #content input[type="url"], #content textarea, #content select { background-color: #fafafa; }
+input.date, input[type="text"], input[type="password"], input[type="email"], input[type="number"], input[type="tel"], input[type="url"], textarea, select { outline: none; padding: 3px 4px; background-color: #fff; border: 1px solid #ccc; box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.075); border-radius: 3px; color: #000; -webkit-transition: all 0.15s ease-in 0s; transition: all 0.15s ease-in 0s; vertical-align: middle; }
+#content input[type="text"], #content input[type="password"], #content input[type="email"], #content input[type="number"], #content input[type="tel"], #content input[type="url"], #content textarea, #content select { background-color: #fafafa; color: #000; }
 input[type="text"]:focus, input[type="password"]:focus, input[type="email"]:focus, input[type="number"]:focus, input[type="tel"]:focus, input[type="url"]:focus, textarea:focus, select:focus { border-color: #51a7e8; box-shadow: inset 0 1px 2px rgba(0, 0, 0, 0.075), 0 0 5px rgba(81, 167, 232, 0.5); }
 
 input.autocomplete { padding-left: 20px; }
```

We also moved from Textile to Markdown for Wiki pages. The line height is massive when previewing, but this is only the case when the header tags (h1, h2. etc) were not converted to the markdown equivalent (\#, \##). Changing them to the markdown equivalent fixes the line heights.

So there nothing to do really. :)

**Comment by makotokw** [comment]
I released v1.2.0 for Redmine 4.0
